docs/7.x/testplan.md

```diff
@@ -429,17 +431,39 @@ root$ gravity resource create cloud-config.yaml --confirm -m
 
 This scenario tests garbage collection on a cluster.
 
-Prerequisites: multi-node cluster with at least 1 regular node. Regular node is necessary to test both master and regular node update paths.
+Prerequisites: multi-node cluster with at least 1 node `--role=knode` and 1 `--role=node|master`.
+Having both `knode` and `node|master` is necessary to test both master and regular node update paths.
 
-Install a previous LTS version, upgrade to the latest version.
+Install a previous LTS version. Gather baseline pre-upgrade state:
 
-After upgrade execute `gravity gc` on the cluster.
+- [ ] `sudo du -sh /var/lib/gravity/{local/packages,site/packages,planet/registry}`
 
-- [ ] Verify the operation completes successfully.
- - [ ] Verify that packages from the previous installation have been removed locally.
- - [ ] Verify that packages from the previous installation have been removed from cluster package storage.
- - [ ] Verify that packages from the current installation are still present.
- - [ ] Tentative: Verify that application packages from remote clusters are still present.
+Upgrade to the release under test. Gather pre-garbage-collection state.
+
+- [ ] `sudo du -sh /var/lib/gravity/{local/packages,site/packages,planet/registry}`
+- [ ] List all images/tags in the registry
+ - `sudo find /var/lib/gravity/planet/registry/ -path '*tags/*' -type d | egrep 'tags/[^/]+$' | sort`
+- [ ] `sudo gravity system gc package --dry-run`
+
+Verify journal logs have been pruned [gravity.e#3429](https://github.com/gravitational/gravity.e/issues/3429)
+ - [ ] `sudo gravity exec du -h /var/log/journal` should show only one subdir.
+
+Execute `gravity gc`. Gather the operation id.
+
+- [ ] Verify the gc operation completes successfully.
+ - `sudo gravity plan --operation-id=<id>`
+- [ ] Verify packages from the previous installation have been removed locally.
+ - `sudo gravity system gc package --dry-run` should no longer list any packages to remove.
+ - `sudo du -sh /var/lib/gravity/local/packages` should show substantially lower usage (similar to baseline).
+- [ ] Verify packages from the previous installation have been removed from cluster package storage.
+ - `sudo du -sh /var/lib/gravity/site/packages` should show substantially lower usage (similar to baseline).
+- [ ] Verify packages from the current installation are still present.
```

Comment:
nit: if a newer version has been uploaded, it will also be still present even if not installed.

Reply:
Is are uploaded packages that are never used something we should explicitly add to this scenario?

If yes: I'll need to rework the instructions earlier to include this, and also do some different expectation setting in the validation (e.g. we wouldn't expect see usage similar to baseline if we still had 2/3 versions present).  If this is what you're asking for, I'd like to handle this as a follow up issue, because it is new testing and I need to focus the rest of this week on wrapping up other scenarios.

If no: In my opinion multiple uploaded versions is far enough off the script that we probably don't need to explicitly mention/handle it.  Anyone tester who is uploading multiple artifacts is 1) not following the instructions and 2) thus probably a power user who is able to reason about the expected state of the package cache and either:
* infer success criteria
* improve these instructions to accommodate their goals

Either way, I'd like further guidance about what you're looking for.  If it is just language, give me the exact wording you want and I'd be glad to include it in this PR.

Reply:
No - I guess I just wanted to have a detailed picture of the gc on paper but we don't really need this as an additional test step/requirement.

Reply:
I think Roman's suggestions in https://github.com/gravitational/gravity/pull/1851#discussion_r452939999 will help with this.  I'm evaluating those commands presently.